### PR TITLE
Local server Sample to allow packages to be changed

### PR DESCRIPTION
**Pull request by vGandhi**
https://devtopia.esri.com/runtime/java-se-sdk/issues/993

@john0005 Please review.

* Added a button to select a package file from the system.

### CODE REVIEW

File: src/main/java/com/esri/samples/localserver/local_server_services/LocalServerServicesController.java [hunk 1]
```diff
@@ -123,6 +135,32 @@ private void handleStartSelectedService() {
...
+        lblSelectedPackage.setText(pathStart +  "MessageInABottle.gpk");
+        break;
+      default:
+        break;
+    }
+  }
+  
+  @FXML
+  private void handleSelectPackage() {
+    File selectedFile = LocalServerServicesSample.openFileDialog();
```
Comment (ghost) [inline src/main/java/com/esri/samples/localserver/local_server_services/LocalServerServicesController.java]: You can get the stage from the scene like so: 
```java
File selectedFile = new FileChooser().showOpenDialog(btnStartServer.getScene().getWindow());
```
No need for the static method and field on the main class.

File: src/main/resources/fxml/local_server_services.fxml [hunk 2]
```diff
@@ -27,20 +28,24 @@
...
+            <items>
+              <FXCollections fx:factory="observableArrayList">
+                <String fx:value="Map Service"/>
+                <String fx:value="Feature Service"/>
+                <String fx:value="Geoprocessing Service"/>
+              </FXCollections>
+            </items>
+          </ComboBox>
+          <Button fx:id="btnSelectPackage" text="Select package" disable="${!btnStartServer.disable}" onAction="#handleSelectPackage"/>
+          <Label fx:id="lblSelectedPackage" text=""/>
```
Comment (ghost) [inline src/main/resources/fxml/local_server_services.fxml]: text attribute not needed as it is set to the default value

File: src/main/java/com/esri/samples/localserver/local_server_services/LocalServerServicesController.java [hunk 3]
```diff
@@ -18,19 +18,31 @@
...
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListView;
+import javafx.scene.control.TextArea;
 
 public class LocalServerServicesController {
 
   @FXML private Button btnStartServer;
   @FXML private ComboBox<String> serviceOptions;
+  @FXML private Button btnSelectPackage;
```
Comment (ghost) [inline src/main/java/com/esri/samples/localserver/local_server_services/LocalServerServicesController.java]: I prefer the naming convention: `selectPackageButton`. Most of our samples use this convention.

Comment (ghost) [reply src/main/java/com/esri/samples/localserver/local_server_services/LocalServerServicesController.java]: Also, you can remove this line for `btnSelectPackage` as it is not used in the code.

**Review by ghost — COMMENTED** [review]
I left a few suggestions. Also, I know you didn't write this part. But I don't think we need the check to see if the service has already started. Local Server seems fine making copies of the same service and I think we should allow/demonstrate that.

File: src/main/java/com/esri/samples/localserver/local_server_services/LocalServerServicesController.java (hunk 3 above)
Comment (vGandhi) [reply src/main/java/com/esri/samples/localserver/local_server_services/LocalServerServicesController.java]: Updated name.

File: src/main/java/com/esri/samples/localserver/local_server_services/LocalServerServicesController.java (hunk 1 above)
Comment (vGandhi) [reply src/main/java/com/esri/samples/localserver/local_server_services/LocalServerServicesController.java]: Yep, that's better.

File: src/main/resources/fxml/local_server_services.fxml (hunk 2 above)
Comment (vGandhi) [reply src/main/resources/fxml/local_server_services.fxml]: Removed.

**Comment by vGandhi** [comment]
@tyle8552 I have addressed your feedback.

**Comment by john0005** [comment]
Looks good 👍

**Pull request merged**